extensions/PhabBugz/lib/WebService.pm

```diff
@@ -0,0 +1,135 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::PhabBugz::WebService;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base qw(Bugzilla::WebService);
+
+use Bugzilla::Attachment;
+use Bugzilla::Bug;
+use Bugzilla::BugMail;
+use Bugzilla::Error;
+use Bugzilla::User;
+use Bugzilla::Util qw(correct_urlbase detaint_natural);
+use Bugzilla::WebService::Constants;
+
+use Bugzilla::Extension::PhabBugz::Util qw(
+    create_private_revision_policy
+    edit_revision_policy
+    get_project_phid
+    intersect
+    make_revision_public
+    request
+);
+
+use Data::Dumper;
+
+use constant PUBLIC_METHODS => qw(
+    revision
+);
+
+sub revision {
+    my ($self, $params) = @_;
+
+    (defined $params->{revision} && detaint_natural($params->{revision}))
+        || ThrowCodeError('param_required', { param => 'revision' });
+
+    my $user = Bugzilla->set_user(Bugzilla::User->new({ name => 'conduit@mozilla.bugs' }));
+
+    # Obtain more information about the revision from Phabricator
+    my $revision_id = $params->{revision};
+    my $result = request('differential.revision.search', {
+        queryKey => 'active',
+        constraints => {
+            ids => [ int($revision_id) ]
+        }
+    });
+
+    (exists $result->{result}{data} && @{ $result->{result}{data} })
```

Comment:
Here postfix unless is definitely preferred, as we're not performing any action on $result.

```perl
ThrowUserError('invalid_phabricator_revision_id') 
    unless exists $result->{result}{data} && @{ $result->{result}{data} });
```